cpp/src/io/orc/stripe_init.cu

```diff
@@ -226,6 +226,26 @@ enum row_entry_state_e {
   STORE_INDEX2,
 };
 
+/**
+ * @brief Calculates the order of index streams based on the index types present in the column.
+ *
+ * @param index_types_bitmap The bitmap of index types showing which index streams are present
+ *
+ * @return The order of index streams
+ */
+static auto __device__ index_order_from_index_types(uint32_t index_types_bitmap)
+{
+  constexpr std::array full_order = {CI_PRESENT, CI_DATA, CI_DATA2};
+
+  std::array<uint32_t, full_order.size()> partial_order;
+  auto curr_idx = 0;
+  for (auto index_type : full_order) {
+    // If the index type is present, add it to the partial order
+    if (index_types_bitmap & (1 << index_type)) { partial_order[curr_idx++] = index_type; }
+  }
+  return partial_order;
```

Comment:
Could you use `std::copy_if` instead here? It is order-preserving.

Reply:
great suggestion, done.